### PR TITLE
Fix Python37 PostCommit azure-cli not support Bookworm

### LINKED ISSUE #27019: [Failing Test]: Azure Integration test is failing in python 3.7 PostCommit
### What happened?

gradle task: `:sdks:python:test-suites:direct:py37:azureIntegrationTest`

Stack trace
```
> Task :sdks:python:test-suites:direct:py37:azureIntegrationTest
azurite_1  | 172.25.0.3 - - [05/Jun/2023:13:22:38 +0000] "PUT /devstoreaccount1/container?restype=container HTTP/1.1" 201 -
test_1     | {
test_1     |   "created": true
test_1     | }
test_1     | azure_integration_test run-test-pre: commands[3] | az storage blob upload -f kinglear.txt -c container -n kinglear.txt --connection-string 'DefaultEndpointsProtocol=https;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=https://azurite:10000/devstoreaccount1;'
test_1     | WARNING: test command found but not installed in testenv
test_1     |   cmd: /usr/bin/az
test_1     |   env: /app/sdks/python/target/.tox/azure_integration_test
test_1     | Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
test_1     | 
test_1     | DEPRECATION WARNING: this will be an error in tox 4 and above!
azurite_1  | 172.25.0.3 - - [05/Jun/2023:13:22:40 +0000] "PUT /devstoreaccount1/container/kinglear.txt HTTP/1.1" 201 -
test_1     | 
Alive[################################################################]  100.0000%
Finished[#############################################################]  100.0000%
test_1     | {
test_1     |   "client_request_id": "0bae5a4a-03a4-11ee-8aa1-0242ac190003",
test_1     |   "content_md5": "38OlDCww42iqPkiLJcxRDg==",
test_1     |   "date": "2023-06-05T13:22:40+00:00",
test_1     |   "encryption_key_sha256": null,
test_1     |   "encryption_scope": null,
test_1     |   "etag": "\"0x1AE0AB209AF41A0\"",
test_1     |   "lastModified": "2023-06-05T13:22:40+00:00",
test_1     |   "request_id": "dfdd10ff-3d9b-4dcb-a2df-d86b57361d62",
test_1     |   "request_server_encrypted": true,
test_1     |   "version": "2022-11-02",
test_1     |   "version_id": null
test_1     | }
test_1     | azure_integration_test run-test: commands[0] | python -m apache_beam.examples.wordcount --input 'azfs://devstoreaccount1/container/*' --output azfs://devstoreaccount1/container/py-wordcount-integration --azure_connection_string 'DefaultEndpointsProtocol=https;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=https://azurite:10000/devstoreaccount1;'
test_1     | INFO:root:Missing pipeline option (runner). Executing pipeline using the default runner: DirectRunner.
test_1     | INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'https://azurite:10000/devstoreaccount1/container?restype=REDACTED&comp=REDACTED&prefix=REDACTED'
test_1     | Request method: 'GET'
test_1     | Request headers:
test_1     |     'x-ms-version': 'REDACTED'
test_1     |     'Accept': 'application/xml'
test_1     |     'User-Agent': 'azsdk-python-storage-blob/12.17.0b1 Python/3.7.16 (Linux-5.15.0-1032-gcp-x86_64-with-debian-11.7)'
test_1     |     'x-ms-date': 'REDACTED'
test_1     |     'x-ms-client-request-id': '0cbe8d9c-03a4-11ee-83c9-0242ac190003'
test_1     |     'Authorization': 'REDACTED'
test_1     | No body was attached to the request
azurite_1  | 172.25.0.3 - - [05/Jun/2023:13:22:41 +0000] "GET /devstoreaccount1/container?restype=container&comp=list&prefix= HTTP/1.1" 400 -
test_1     | INFO:azure.core.pipeline.policies.http_logging_policy:Response status: 400
test_1     | Response headers:
test_1     |     'Server': 'Azurite-Blob/3.23.0'
test_1     |     'x-ms-error-code': 'InvalidHeaderValue'
test_1     |     'x-ms-request-id': '55c7ab06-f796-49db-974f-a0cc5ae045a8'
test_1     |     'content-type': 'application/xml'
test_1     |     'Date': 'Mon, 05 Jun 2023 13:22:41 GMT'
test_1     |     'Connection': 'keep-alive'
test_1     |     'Keep-Alive': 'REDACTED'
test_1     |     'Transfer-Encoding': 'chunked'
test_1     | Traceback (most recent call last):
test_1     |   File "/usr/local/lib/python3.7/runpy.py", line 193, in _run_module_as_main
test_1     |     "__main__", mod_spec)
test_1     |   File "/usr/local/lib/python3.7/runpy.py", line 85, in _run_code
test_1     |     exec(code, run_globals)
test_1     |   File "/app/sdks/python/apache_beam/examples/wordcount.py", line 111, in <module>
test_1     |     run()
test_1     |   File "/app/sdks/python/apache_beam/examples/wordcount.py", line 90, in run
test_1     |     lines = p | 'Read' >> ReadFromText(known_args.input)
test_1     |   File "/app/sdks/python/apache_beam/io/textio.py", line 790, in __init__
test_1     |     escapechar=escapechar)
test_1     |   File "/app/sdks/python/apache_beam/io/textio.py", line 144, in __init__
test_1     |     validate=validate)
test_1     |   File "/app/sdks/python/apache_beam/io/filebasedsource.py", line 127, in __init__
test_1     |     self._validate()
test_1     |   File "/app/sdks/python/apache_beam/options/value_provider.py", line 193, in _f
test_1     |     return fnc(self, *args, **kwargs)
test_1     |   File "/app/sdks/python/apache_beam/io/filebasedsource.py", line 188, in _validate
test_1     |     match_result = FileSystems.match([pattern], limits=[1])[0]
test_1     |   File "/app/sdks/python/apache_beam/io/filesystems.py", line 204, in match
test_1     |     return filesystem.match(patterns, limits)
test_1     |   File "/app/sdks/python/apache_beam/io/filesystem.py", line 804, in match
test_1     |     raise BeamIOError("Match operation failed", exceptions)
test_1     | apache_beam.io.filesystem.BeamIOError: Match operation failed with exceptions {'azfs://devstoreaccount1/container/*': BeamIOError('List operation failed with exceptions {\'azfs://devstoreaccount1/container/\': HttpResponseError(\'The API version 2023-01-03 is not supported by Azurite. Please upgrade Azurite to latest version and retry. If you are using Azurite in Visual Studio, please check you have installed latest Visual Studio patch. Azurite command line parameter "--skipApiVersionCheck" or Visual Studio Code configuration "Skip Api Version Check" can skip this error. \\nRequestId:55c7ab06-f796-49db-974f-a0cc5ae045a8\\nTime:2023-06-05T13:22:41.855Z\\nErrorCode:InvalidHeaderValue\')}')}
test_1     | ERROR: InvocationError for command /app/sdks/python/target/.tox/azure_integration_test/bin/python -m apache_beam.examples.wordcount --input 'azfs://devstoreaccount1/container/*' --output azfs://devstoreaccount1/container/py-wordcount-integration --azure_connection_string 'DefaultEndpointsProtocol=https;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=https://azurite:10000/devstoreaccount1;' (exited with code 1)
test_1     | azure_integration_test run-test-post: commands[0] | bash /app/sdks/python/scripts/run_tox_cleanup.sh
test_1     | WARNING: test command found but not installed in testenv
test_1     |   cmd: /bin/bash
test_1     |   env: /app/sdks/python/target/.tox/azure_integration_test
test_1     | Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
test_1     | 
test_1     | DEPRECATION WARNING: this will be an error in tox 4 and above!
test_1     | ___________________________________ summary ____________________________________
test_1     | ERROR:   azure_integration_test: commands failed
```

### Issue Failure

Failure: Test is continually failing

### Issue Priority

Priority: 1 (unhealthy code / failing or flaky postcommit so we cannot be sure the product is healthy)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by Abacn**
Fixes #27019

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Abacn** [comment]
#26988 Fixed website precommit failure due to debian bookworm dropped support of Node 10. Fixed by upgrading Node->14

**Comment by Abacn** [comment]
Run Python 3.7 PostCommit

**Comment by Abacn** [comment]
Run Python 3.7 PostCommit

**Comment by Abacn** [comment]
test passed, ready for review now

**Comment by Abacn** [comment]
R: @AnandInguva @tvalentyn

File: sdks/python/tox.ini [hunk 1]
```diff
@@ -210,6 +210,7 @@ passenv = REQUESTS_CA_BUNDLE
 allowlist_externals = 
   wget
   az
+  bash
```
Comment (AnandInguva) [inline sdks/python/tox.ini:213]: why do we need bash here?

Comment (AnandInguva) [reply sdks/python/tox.ini:213]: Okay. looks like `commands_post` use bash and the `allowlist_externals` in the current context overwrites the `allowlist_externals` in the global context of tox.ini

**Review by AnandInguva — APPROVED** [review]

**Comment by Abacn** [comment]
Test passed: https://ci-beam.apache.org/job/beam_PostCommit_Python37_PR/549/

**Pull request merged**